### PR TITLE
feat(@schematics/angular): add solutions style tsconfig structure

### LINKED ISSUE #8138: tsconfig paths and module resolution errors in VS Code
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

> @angular/cli: 1.4.9
node: 7.9.0
os: darwin x64
@angular/animations: 4.4.5
@angular/common: 4.4.5
@angular/compiler: 4.4.5
@angular/core: 4.4.5
@angular/forms: 4.4.5
@angular/http: 4.4.5
@angular/platform-browser: 4.4.5
@angular/platform-browser-dynamic: 4.4.5
@angular/router: 4.4.5
@angular/cli: 1.4.9
@angular/compiler-cli: 4.4.5
typescript: 2.3.4

macOS Sierra 10.12.6
VS Code 1.17.2
### Repro steps.

My tsconfig path property seems broken.
When I replace my path mapped imports (@shared, @assets etc.) with the equivalent coded relative paths, typescript module resolution seems to work (vs code intellisense is happy)
I have a large angular 2 project and am used to aliasing my asset and shared directories this way.
NOTE: project builds and (seems) to run just fine
This has been reported often, but I can't make sense of the current state of this issue.
This more than annoying as VS Code reports Angular spurious errors on all of my components.
I also suspect this is why I get the infamous Can't read property isSkipSelf of null when I open my templates (html) in VS Code.
I was running 1.4.2 and just tried ~1.4.2, which resolves to 1.4.9 in my package.json, to no avail.
Will 1.5.x fix this?
How can I help debug this without providing a copy of my app?

Thanks in advance. Regards.

### LINKED ISSUE #17493: Wrong jasmine/wd2 typings in Angular 8/9
I created a new blank angular 9 project using next commands:
```
ng new my-app
npm install
ng serve
```

I added `then()` to test:

Then run:

```
ng test
```

Error occured:

```
ERROR in src/app/app.component.spec.ts:20:30 - error TS2339: Property 'then' does not exist on type 'boolean'.

20     expect(app).toBeTruthy().then(() => { });
                                ~~~~
```

After removing **"@types/jasminewd2": "~2.0.3",** from **package.json**, deleting **then()** of course and running:

```
npm install
ng test // no compilation error
```

Visual studio code reported that expect() matcher is **sync method** and returns boolean instead of Promise, just removed the .then() and everything works as it should.
```
## my environment

**Angular Version:**
Angular CLI: 9.1.1
Node: 12.16.1
OS: win32 x64

Angular: 9.1.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.1
@angular-devkit/build-angular     0.901.1
@angular-devkit/build-optimizer   0.901.1
@angular-devkit/build-webpack     0.901.1
@angular-devkit/core              9.1.1
@angular-devkit/schematics        9.1.1
@angular/cli                      9.1.1
@ngtools/webpack                  9.1.1
@schematics/angular               9.1.1
@schematics/update                0.901.1
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

D:\my-app>node -v
v12.16.1

D:\my-app>npm -v
6.13.4
```
Not really sure this is an Angular bug, but package @types/jasminewd2 comes by default I suppose.

### PR DESCRIPTION
**Pull request by alan-agius4**
In version 3.9, TypeScript introduced the concept of "Solutions Style" tsconfig to improve developer experience.

More info: https://devblogs.microsoft.com/typescript/announcing-typescript-3-9-rc/#solution-style-tsconfig

Closes #17493 and closes #8138

### CODE REVIEW

**Comment by alan-agius4** [comment]
Blocking until support for TS 3.9 is landed in the framework. None the less, we should get this in a ready to be merged when that happens.

File: tests/legacy-cli/e2e/utils/project.ts [hunk 1]
```diff
@@ -39,10 +39,10 @@ export async function createProject(name: string, ...args: string[]) {
   await ng('new', name, '--skip-install', ...extraArgs, ...args);
   process.chdir(name);
 
-  if (fs.existsSync('tsconfig.json')) {
+  if (fs.existsSync('tsconfig.base..json')) {
```
Comment (clydin) [inline tests/legacy-cli/e2e/utils/project.ts:42]:
```suggestion
  if (fs.existsSync('tsconfig.base.json')) {
```

Comment (alan-agius4) [reply tests/legacy-cli/e2e/utils/project.ts]: good catch

**Comment by kyliau** [comment]
We had a discussion whether the filename convention should be `tsconfig-base.json` or `tsconfig.base.json`. The former follows the same pattern used in the TS repo, whereas the latter allows the Angular workspace to enforce a consistent file naming. 
Another concern is the convention used in google3. However, since google3 **does not** have plan to introduce solution-style tsconfig, the consideration is moot here.
We've decided to with `tsconfig.base.json` to ensure a consistent naming across Angular workspace.
cc: @IgorMinar

File: integration/angular_cli/tsconfig.base.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "typeRoots": [
+      "node_modules/@types"
+    ],
+    "lib": [
+      "es2017",
+      "dom"
+    ]
+  },
+  "angularCompilerOptions": {
+    "enableIvy": true,
```
Comment (IgorMinar) [inline integration/angular_cli/tsconfig.base.json:22]: this is the default. no? Why not remove it?

Comment (alan-agius4) [reply integration/angular_cli/tsconfig.base.json:22]: yeah, but for the integration tests this is done to easily switch between true and false.

Comment (IgorMinar) [reply integration/angular_cli/tsconfig.base.json:22]: ah. ok

**Comment by IgorMinar** [comment]
> We've decided to with `tsconfig.base.json` to ensure a consistent naming across Angular workspace.

Sounds great! thanks for documenting the decision making process.

**Pull request merged**